src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -755,6 +755,13 @@ struct BlockZipperJoinImpl {
   CompatibleRowAction& compatibleRowAction_;
   [[no_unique_address]] IsUndef isUndefined_{};
 
+  using LeftBlocks = typename LeftSide::CurrentBlocks;
+  using RightBlocks = typename RightSide::CurrentBlocks;
+
+  // We can't define aliases for these concepts, so we use macros instead.
+#define Side SameAsAny<LeftSide, RightSide> auto
+#define Blocks SameAsAny<LeftBlocks, RightBlocks> auto
```

Comment:
To be ultra safe, 
pleas do an `#if defined(Side) || defined(Blocks) ERROR(...)` to guarantee that we don't accidentally overwrite any macros and introduce bugs that are impossibly to trace....